### PR TITLE
Use headless service for node discovery

### PR DESCRIPTION
**Pull request by proddata**
## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/deployment/containers/kubernetes.rst [hunk 1]
```diff
@@ -129,9 +129,9 @@ Here's an example configuration snippet:
       labels:
         app: crate
     spec:
-      # A static IP address is assigned to this service. This IP address is
-      # only reachable from within the Kubernetes cluster.
-      type: ClusterIP
```
Comment (SStorm) [inline docs/deployment/containers/kubernetes.rst:134]: You will still need the type. What makes a headless service is the `clusterIP: None` thing.

Comment (proddata) [reply docs/deployment/containers/kubernetes.rst:134]: ClusterIP is the default type.
In most definition I found it was omitted for headless services.
The comment is definitely wrong

Comment (SStorm) [reply docs/deployment/containers/kubernetes.rst:134]: Ah yes you're right, just tried it and it works fine w/o `type`

**Review by SStorm — APPROVED** [review]

**Comment by protosam** [comment]
I think the changes should look more like this: https://github.com/protosam/crate-howtos/blob/master/docs/deployment/containers/kubernetes.rst

I would just make a PR, but the contributor's agreement needs amendments before I could agree to it. Will discuss that in the community discuss though.

**Pull request merged**